src/rest/commands/general/follow.ts

```diff
@@ -0,0 +1,100 @@
+import Command, {
+  basicInteractionResponse,
+} from "../../common/command/Command.ts";
+import { bot } from "../../cache.ts";
+import { enTranslate, translate } from "../../common/util/i18next.ts";
+import {
+  DiscordApplicationCommandOptionTypes,
+  followChannel,
+  getChannelWebhooks,
+  snowflakeToBigint,
+  validatePermissions,
+} from "../../../../deps.ts";
+
+const FollowCommand: Command = async (interaction) => {
+  const guildId: bigint = snowflakeToBigint(interaction.guildId!);
+
+  if (
+    !validatePermissions(snowflakeToBigint(interaction.member!.permissions), [
```

Comment:
It appears that `validatePermissions` only checks server-level permissions, not channel-level permissions. I think this should be refactored to check permissions at the channel level (which would inherit the server-level permissions).

Reply:
Would prefer to fix this in a separate PR that covers all commands if that is the case.